lib/active_merchant/billing/gateways/epay.rb

```diff
@@ -158,6 +160,16 @@ def add_instant_capture(post, option)
         post[:instantcapture] = option ? 1 : 0
       end
 
+      def add_3ds_auth(post, options)
+        if options[:three_d_secure]
+          post[:eci] = options.dig(:three_d_secure, :eci)
+          post[:xid] = options.dig(:three_d_secure, :xid)
+          post[:cavv] = options.dig(:three_d_secure, :cavv)
+          post[:threeds_version] = options.dig(:three_d_secure, :version)
+          post[:ds_transaction_id] = options.dig(:three_d_secure, :ds_transaction_id)
```

Comment:
In many cases `xid` and `ds_transaction_id` are exclusive. Would it be ok to have one of these values being `nil`?

Reply:
Yes.